robin.js

```diff
@@ -0,0 +1,852 @@
+  },
+  _onRefresh: function () {
+    var e = Math.random() * 300 * 1e3;
+    setTimeout(function () {
+      location.reload()
+    }, e)
+  },
+  _onClose: function (e) {
+    if (this._connectionAttempts < this._maximumRetries) {
+      var t = this._backoffTime * Math.pow(2, this._connectionAttempts), n = Math.random() * this._retryJitterAmount - this._retryJitterAmount / 2, i = Math.round(t + n);
+      r.debug("websocket: connection lost (" + e.code + "), reconnecting in " + i + "ms"), r.debug("(can't connect? Make sure you've allowed https access in your browser.)"), this.trigger("reconnecting", i), setTimeout(_.bind(this._connect, this), i)
+    } else r.debug("websocket: maximum retries exceeded. bailing out"), this.trigger("disconnected");
+    this._sendStats({websocketError: {error: 1}})
+  }
+}), r.templates.set([{
+  style: "html",
+  name: "robin/robinmessage",
+  template: '<div class="robin-message\n        <% if (thing.displayCompact) { %>\n            robin-message--display-compact\n        <% } %>\n        robin--flair-class--<%- thing.flairClass %>\n        robin--message-class--<%- thing.messageClass %>\n        robin--user-class--<%- thing.userClass %>">\n    <time class="robin-message--timestamp" datetime="<%- thing.isoDate %>"><%- thing.timestamp %></time>\n    <% if (!thing.displayCompact) { %>\n      <span class="robin-message--from robin--username"><%- thing.author %></span>\n    <% } %>\n    <span class="robin-message--message"><%- thing.message %></span>\n</div>\n'
+}]), r.templates.set([{
+  style: "html",
+  name: "robin/robinroomparticipant",
+  template: '<div class="robin-room-participant\n        robin--user-class--<%- thing.userClass %>\n        robin--presence-class--<%- thing.presenceClass %>\n        robin--vote-class--<%- thing.voteClass %>">\n    <span class="robin--icon"></span>\n    <span class="robin--username"><%- thing.from %></span>\n</div>\n'
+}]), !function (e) {
+  function n() {
+  }
+
+  function r(e) {
+    var t = e.stack;
+    return t && t.split("\n").slice(1).join("\n")
+  }
+
+  function i(e, t, n, s) {
+    if (!(this instanceof i))return new i(e, t, n, s);
+    var o = Error.call(this);
+    if ("stack" in o)if ("captureStackTrace" in Error)Error.captureStackTrace(this, i); else try {
+      Object.defineProperty(this, "stack", {
+        configurable: !0, get: function () {
+          var e = r(o);
+          return this.stack = e
+        }
+      })
+    } catch (u) {
+      this.stack = r(o)
+    }
+    var a = e + " | " + t;
+    this.name = "ApiError", this.message = a, this.displayName = e, this.displayMessage = t, this.field = n || "", this.source = s || "api"
+  }
+
+  var t = {
+    UNKNOWN_ERROR: e._("unknown error %(message)s"),
+    NO_TEXT: e._("we need something here"),
+    TOO_LONG: e._("this is too long (max: %(max_length)s)"),
+    TOO_SHORT: e._("this is too short (min: %(min_length)s)"),
+    SR_RULE_EXISTS: e._("A subreddit rule by that name already exists."),
```

Comment:
This is interesting. Can track this down to figure out how a subreddit gets created.